linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1013,8 +1027,8 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
                 NewArray(tpe, recs(lengths, allowUnpure))
               case ArrayValue(tpe, elems) =>
                 ArrayValue(tpe, recs(elems, allowUnpure))
-              case JSArrayConstr(items) if !containsAnySpread(items) =>
-                JSArrayConstr(recs(castNoSpread(items), allowUnpure))
+              case JSArrayConstr(items) if items.forall(_.isInstanceOf[Tree]) =>
+                JSArrayConstr(recs(items.asInstanceOf[List[Tree]], allowUnpure))
```

Comment:
Humm... This is not nice. IIUC, this basically means we cannot always keep spreads in `JSArrayConstr`s in expression position. Maybe this warrants a comment?

Reply:
I managed to fix that. It wasn't that hard, actually.